sync.js

```diff
@@ -20,181 +17,120 @@ const SYNCFILE_FORMATS = {
   'osm-p2p-syncfile'   : 2
 }
 
+const DEFAULT_OPTS = {
+  dns: {
+    interval: 3000
+  },
+  dht: false
+}
+
+function WifiTarget (peer) {
+  peer.type = 'wifi'
+  return peer
+}
+
+function FileTarget (filename) {
+  return {
+    id: path.basename(filename),
+    filename,
+    type: 'file'
+  }
+}
+
 class Sync extends events.EventEmitter {
   constructor (osm, media, opts) {
     super()
-    if (!opts) opts = {}
+    this.opts = Object.assign(DEFAULT_OPTS, opts)
     opts.writeFormat = opts.writeFormat || 'hyperlog-sneakernet'
     if (!SYNCFILE_FORMATS[opts.writeFormat]) throw new Error('unknown syncfile write format: ' + opts.writeFormat)
 
     this.osm = osm
     this.media = media
-    this.host = opts.host
-    this.id = opts.id || 'Mapeo_' + randombytes(8).toString('hex')
+    this.id = opts.id || randombytes(8).toString('hex')
     this.opts = opts
+
+    // track replication progress states of files and wifi streams
     this._targets = {}
-    this._replicationServer = null
+
+    this.swarm = null
+  }
+
+  _swarm () {
+    this.swarm = Swarm()
+    this.swarm.on('connection-closed', (connection, info) => {
+      const target = WifiTarget(info)
+      this.emit('down', target)
```

Comment:
still emitting `down` for api-compat but we can change this to be `connection-closed` if we want.